### PR TITLE
feat: added rudimentary login screen

### PR DESCRIPTION
**Pull request by sulemanmalik**

### CODE REVIEW

File: src/Features/Login/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: TypeScript interfaces already have type checking for everything. With TS, we don't need the `prop-types` package at all. Should uninstall this.

File: src/Features/Login/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import classNames from 'classnames';
+import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: Unused import

**Review by erickzhao — CHANGES_REQUESTED** [review]
A few small things

Also this doesn't compile

File: src/Features/Login/Login.tsx [hunk 3]
```diff
@@ -1,101 +1,86 @@
...
-                            label="Password"
-                            className={classNames(classes.textField, classes.dense)}
-                            margin="dense"
-                            variant="outlined"
-                        />  
-                    <Button size="medium" className={classes.margin} color="primary" variant="contained">Sign In</Button>
-                    </Grid>
+class Login extends React.Component<IProps, IState> {
+  public state = {
+    email: 'test@gmail.com',
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: Why initialize the state to these placeholder values? Should keep as empty strings.

File: src/Features/Login/Login.tsx [hunk 4]
```diff
@@ -1,101 +1,86 @@
...
-        margin: theme.spacing.unit
+    card: {
+      alignItems: 'center',
+      display: 'flex',
+      justifyContent: 'center',
+      paddingBottom: 12,
+      paddingLeft: 12,
+      paddingTop: 12,
     },
     container: {
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: What does the `.container` class do? Removing it doesn't affect the appearance of the component.

File: src/Features/Login/Login.tsx [hunk 5]
```diff
@@ -1,101 +1,86 @@
...
-import Button from '@material-ui/core/Button';
-import Card from '@material-ui/core/Card';
-import { CardContent } from '@material-ui/core';
+import React from 'react';
 
 const styles = (theme: Theme) =>
   createStyles({
-    margin: {
-        margin: theme.spacing.unit
+    card: {
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: What is `.card` doing? All I see is that it's adding some uneven padding.
screencast 2019-02-05 21-19-05

File: src/Features/Login/Login.tsx [hunk 6]
```diff
@@ -1,101 +1,86 @@
...
+    const { classes } = this.props;
+    return (
+      <form className={classes.container} noValidate autoComplete="off">
+        <Grid
+          container
+          spacing={0}
+          direction="column"
+          alignItems="center"
+          justify="center"
+          style={{ minHeight: '100vh' }}
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: Use the `styles` thing instead of injecting inline css directly.

File: tslint.json [hunk 7]
```diff
@@ -5,6 +5,8 @@
         "tslint-config-prettier"
     ],
     "jsRules": {},
-    "rules": {},
+    "rules": {
+        "no-implicit-dependencies": false
```
Comment (erickzhao) [inline tslint.json]: Why are we disabling this?

**Review by erickzhao — CHANGES_REQUESTED** [review]
More changes needed

File: src/Features/Login/Login.tsx (hunk 1 above)
Comment (erickzhao) [reply src/Features/Login/Login.tsx]: @sulemanmalik still need to `npm uninstall prop-types`

File: src/Features/Login/Login.tsx (hunk 3 above)
Comment (sulemanmalik) [reply src/Features/Login/Login.tsx]: Changed to empty strings.

File: src/Features/Login/Login.tsx (hunk 4 above)
Comment (sulemanmalik) [reply src/Features/Login/Login.tsx]: Removed container, I thought it was required to have it as a child of card.

File: src/Features/Login/Login.tsx (hunk 5 above)
Comment (sulemanmalik) [reply src/Features/Login/Login.tsx]: Removed `.card`

File: tslint.json (hunk 7 above)
Comment (sulemanmalik) [reply tslint.json]: This was giving me an issue with `import classNames from 'classnames'` in `Login.tsx` line 2 and this was the solution I saw on stack overflow.

File: src/Features/Login/Login.tsx (hunk 6 above)
Comment (sulemanmalik) [reply src/Features/Login/Login.tsx]: done

File: tslint.json (hunk 7 above)
Comment (erickzhao) [reply tslint.json]: The issue is that we don't have `classnames` as a dependency in our `package.json`. That's what the warning called out. Maybe add that instead and add the rule back?

Comment (sulemanmalik) [reply tslint.json]: Weird thing is that `classnames` is in the `package.json` so I was confused

File: src/App.tsx [hunk 8]
```diff
@@ -1,26 +1,11 @@
 import React, { Component } from "react";
-import logo from "./logo.svg";
-
-import "./App.css";
+import Login from './Features/Login/Login';
 
 class App extends Component {
   public render() {
     return (
       <div className="App">
```
Comment (erickzhao) [inline src/App.tsx]: Don't need `className` since you removed the associated CSS file

File: tslint.json [hunk 9]
```diff
@@ -5,6 +5,8 @@
         "tslint-config-prettier"
     ],
     "jsRules": {},
-    "rules": {},
+    "rules": {
```
Comment (erickzhao) [inline tslint.json]: Reformat

File: src/Features/Login/Login.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+              </Grid>
+            </CardContent>
+          </Card>
+        </Grid>
+      </form>
+    )
+  }
+}
+
+export default withStyles(styles)(Login);
```
Comment (erickzhao) [inline src/Features/Login/Login.tsx]: Can you add a newline at end of file? You can make this happen easily with VSCode settings

image

**Review by erickzhao — CHANGES_REQUESTED** [review]
A few more nitpicks

**Review by erickzhao — APPROVED** [review]

**Pull request merged**